### PR TITLE
More categories

### PR DESCRIPTION
**Pull request by J3fftw1**
## Short Description
First step into making more categories within 1 category 

## Additions/Changes/Removals
Added a flexCategory

## Related Issues
N/A
## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

**Comment by J3fftw1** [comment]
i would like some help on this for anyone seeing this i have no clue how to register an category to this.

File: src/main/java/dev/j3fftw/litexpansion/catergory/MainCategory.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import me.mrCookieSlime.Slimefun.cscorelib2.item.CustomItem;
+import org.bukkit.ChatColor;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+public class MainCategory extends FlexCategory {
+
+    private ChestMenu menu;
```
Comment (Mooy1) [inline src/main/java/dev/j3fftw/litexpansion/catergory/MainCategory.java:20]: Move this to Open(), my pr has example

File: src/main/java/dev/j3fftw/litexpansion/catergory/MainCategory.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    public MainCategory() {
+        super(new NamespacedKey(LiteXpansion.getInstance(), "test_category"),
+            new CustomItem(Items.GENERATOR, "&5test")
+        );
+
+        setupInv();
+        menu.setEmptySlotsClickable(false);
+    }
+
+    private void setupInv() {
```
Comment (Mooy1) [inline src/main/java/dev/j3fftw/litexpansion/catergory/MainCategory.java:33]: Also move this to Open, or add a ChestMenu field to it and just move the new ChestMenu

File: src/main/java/dev/j3fftw/litexpansion/catergory/MainCategory.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    @Override
+    public boolean isVisible(Player p, PlayerProfile profile, SlimefunGuideLayout layout) {
+        return true;
+    }
+
+    @Override
+    public void open(Player player, PlayerProfile playerProfile, SlimefunGuideLayout slimefunGuideLayout) {
+        menu.addItem(1, new CustomItem(ChestMenuUtils.getBackButton(player, "",
+            ChatColor.GRAY + SlimefunPlugin.getLocalization().getMessage(player, "guide.back.guide")))
+        );
```
Comment (Mooy1) [inline src/main/java/dev/j3fftw/litexpansion/catergory/MainCategory.java:85]: menu.setEmptySlotsClickable(false);

File: src/main/java/dev/j3fftw/litexpansion/catergory/MainCategory.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        final ChestMenu playerMenu = new ChestMenu(SlimefunPlugin.getLocalization().getMessage(p, "guide.title.main"));
+        playerMenu.setEmptySlotsClickable(false);
+        return playerMenu;
+    }
+
+    private void displayItem(PlayerProfile profile, ItemStack result) {
+        final Player p = profile.getPlayer();
+        if (p != null) {
+            final ChestMenu menu = this.create(p);
+
```
Comment (Mooy1) [inline src/main/java/dev/j3fftw/litexpansion/catergory/MainCategory.java:52]: menu.setEmptySlotsClickable(false);

**Pull request closed without merging**